### ISSUE TITLE
Guidance regarding fresco and memory usage 

### ISSUE THREAD

**Issue by bottiger** [facebook/fresco#311]
I would like some help regarding some best practices when working with Fresco.

I'm developing a [(rather simple podcasting) app](https://github.com/bottiger/SoundWaves/). You can subscribe to podcasts and it can show you a list of recent episodes.

However, I'm struggling with a high memory usage. I know this doesn't need to be a bad thing by itself but it doesn't look good when my (rather simple) app takes up 200 MB in the "Running apps" section.

I did some memory profiling and it seems that 90% of the memory is byte[] objects owned by fresco. (x number of images around 7-8 MB each)

![byte](https://cloud.githubusercontent.com/assets/489129/7796802/95e7dee0-02e3-11e5-9cc3-7fdd046028e4.JPG)
![leak_suspects](https://cloud.githubusercontent.com/assets/489129/7796800/92a23866-02e3-11e5-9374-5d19f25e7124.JPG)

I have an app which displays a list of images, fetch from the web. These (jpg) images are (usually) 1200x1200 in size. In my list they are usually just 50x50 dp. When the user clicks on am image it is expanded to the width of the screen. On a nexus 5 this would be a width of 1080 [but I do resize them to 256x256](http://frescolib.org/docs/resizing-rotating.html#_)

I did some back of the envelope calculations and the numbers seems to match up.

256 * 256 * ( 5 * 6 * 5 ) bytes ~ 10 mb per image

7-9 of these images would quickly consume around 90 mb. If I scroll thought the list I assume this number could grown quickly.

So, how do you suggest developers like me handle this situation? 

1. Should I just leave the app the way it currently is and let the memory consumption stay at it's current level.
2. Should I resize all the images to 50x50 dip during the request and then fetch a new high res version when I need it? (note, this adds some complexity since the high res images is shown in the same SimpleDraweeView. Currently the SimpleDraweeView is just expanded by resizing)
3. Something else?

**TL;DR;** Is a high memory consumption just the nature of Fresco or should I do something about it?

**Secondly**  I have a search bar  where you can search for new podcasts. When performing a search a lot of new images are displayed. This naturally consumes a decent amount of memory. Is there a way to trim the (memory) cache when the search is cleared - since I know these images will probably not be needed any more?

**Comment by sohayb** [comment]
This is scary! I think high memory consumption should be against the nature of Fresco.

**Comment by michalgr** [comment]
10mb per image is a lot more than what I would expect for a single 256x256 pixel image. Why do you multiply number of pixels by 5 * 6 * 5 in your formula ? Single pixel occupies 4 bytes. Let's double check how is this big number produced.

For sure the most efficient thing to do is to ask for an image of exactly the size that you want to display.

When it comes to trimming the cache - you can do that. You can use either this method: https://github.com/facebook/fresco/blob/847851a8de6d223c0a6634447bddee1252e01c64/imagepipeline/src/main/java/com/facebook/imagepipeline/core/ImagePipeline.java#L200, or similar code of your own.
However, I am not sure how much benefit there is in clearing the cache, it probably will be populated with new things soon. I would rather suggest configuring the size of the cache, so that it never goes above a threshold you are comfortable with.

**Comment by tyronen** [comment]
You should map 50x50dp to your screen pixel size and resize to that. 256x256 is likely too large. When you need the 1080x display, you can issue another request without the resize options. This will not download the image again, the original, unresized image remains in disk cache.

**Comment by tyronen** [comment]
Please open a new issue if you need more help.
